beluga/include/beluga/algorithm/distance_map.hpp

```diff
@@ -48,20 +48,22 @@ namespace beluga {
  * \param neighbors_function Given the index i of one cell in the map,
  *  neighbors_function(i) returns the cell indexes of neighbor cells in the
  *  obstacle map.
+ * \param max_distance_value The maximum distance anywhere in the map. Values over this will be truncated to this value.
  * \return A map where each cell value is the distance to the nearest object.
  */
 template <class Range, class DistanceFunction, class NeighborsFunction>
 auto nearest_obstacle_distance_map(
     Range&& obstacle_mask,
     DistanceFunction&& distance_function,
-    NeighborsFunction&& neighbors_function) {
+    NeighborsFunction&& neighbors_function,
+    std::invoke_result_t<DistanceFunction, std::size_t, std::size_t> max_distance_value) {
```

Comment:
@glpuga nit^2: consider making `DistanceType` below a template parameter with default, then use it here.

Reply:
I could not get it to build. If you can provide a version of this that builds, I'll happily take it.

Reply:
Hmm, it's inferring the template parameter from a literal and breaking down. Nevermind.